pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -125,6 +125,11 @@ var (
 			return !wf.Status.FinishedAt.IsZero() && wf.Status.FinishedAt.Time.Before(t)
 		}
 	}
+	WorkflowRanBetween = func(startTime time.Time, endTime time.Time) WorkflowPredicate {
```

Comment:
interesting, I can think of other interpretation of this:

* the workflow started between times A and B
* the workflow finished between times A and. B
* the workflow was running between the two times - i.e. start time is before B and (not finished OR finished after time A)

Reply:
Yes. Should we discuss further on a better design in a separate issue? The original purpose of this PR is to optimize the performance of the existing functionality. We can definitely break this down and see if we can come up something more comprehensive and useful in practice.